### PR TITLE
MRG: Add modified Hughson-Westlake tracker

### PR DESCRIPTION
**Pull request by mpolonenko**
Adding the MWH tracker to the trackers, which is what audiometers use.

### CODE REVIEW

**Comment by larsoner** [comment]
Travis and Azure are not happy (ignore AppVeyor for now), and this will need tests.

**Comment by mpolonenko** [comment]
I've run various tests on the tracker and created an example. It has passed all but 2 tests, and the failures seem unrelated to my code.

File: examples/stimuli/tracker_staircase_MHW.py [hunk 1]
```diff
@@ -0,0 +1,51 @@
+#!/usr/bin/env python3
+# -*- coding: utf-8 -*-
+"""
+Created on Thu Oct 10 12:51:56 2019
+
+@author: mpolonenko
```
[the remark this answers is not in the document]
Comment (mpolonenko) [reply examples/stimuli/tracker_staircase_MHW.py]:
```
#!/usr/bin/env python3
# -*- coding: utf-8 -*-
"""
==============================
Do an adaptive track staircase
==============================

This shows how to make and use an adaptive track using
:class:`expyfun.stimuli.TrackerMHW`.

"""
```

Do you mean keep the first two lines, and have it like this?

**Comment by drammock** [comment]
The errors seem to be unrelated, but the warnings are real:

```
/home/circleci/project/doc/auto_examples/index.rst:494: WARNING: undefined label: sphx_glr_auto_examples_stimuli_tracker_staircase_mhw.py (if the link has no caption the label must precede a section header)
/home/circleci/project/doc/auto_examples/stimuli/sg_execution_times.rst:26: WARNING: undefined label: sphx_glr_auto_examples_stimuli_tracker_staircase_mhw.py (if the link has no caption the label must precede a section header)
generated/expyfun.analyze.sigmoid.examples:68: WARNING: undefined label: sphx_glr_auto_examples_stimuli_tracker_staircase_mhw.py (if the link has no caption the label must precede a section header)
generated/expyfun.analyze.sigmoid.examples:76: WARNING: undefined label: sphx_glr_auto_examples_stimuli_tracker_staircase_mhw.py (if the link has no caption the label must precede a section header)
generated/expyfun.stimuli.TrackerMHW.examples:14: WARNING: undefined label: sphx_glr_auto_examples_stimuli_tracker_staircase_mhw.py (if the link has no caption the label must precede a section header)
generated/expyfun.stimuli.TrackerMHW.examples:22: WARNING: undefined label: sphx_glr_auto_examples_stimuli_tracker_staircase_mhw.py (if the link has no caption the label must precede a section header)
```

Take a look also at how the link to the example is rendering on the test server:
https://671-11614571-gh.circle-artifacts.com/0/html/auto_examples/index.html#stimulus-generation-examples

File: examples/stimuli/tracker_staircase_MHW.py (hunk 1 above)
[the remark this answers is not in the document]
Comment (larsoner) [reply examples/stimuli/tracker_staircase_MHW.py]: Keeping the first two lines is up to you (I would delete the shebang personally), the critical thing is removing the extraneous stuff from the block string

[the remark this answers is not in the document]
Comment (mpolonenko) [reply examples/stimuli/tracker_staircase_MHW.py]: Ok, thanks, I'll delete the shebang as well

File: expyfun/stimuli/tests/test_tracker.py [hunk 2]
```diff
@@ -221,3 +223,64 @@ def test_tracker_dealer():
...
+    tr.threshold
+    tr.stopped
+    tr.x
+    tr.x_current
+    tr.responses
+    tr.n_trials
+    tr.n_reversals
+    tr.reversals
+    tr.reversal_inds
+    tr.threshold_reached
```
[the remark this answers is not in the document]
Comment (mpolonenko) [reply expyfun/stimuli/tests/test_tracker.py]: Thanks, I've made those changes

**Comment by larsoner** [comment]
@mpolonenko now that #403 is merged can you rebase? It will make Travis run properly

**Comment by mpolonenko** [comment]
@larsoner yes, I pushed the changes you suggested, and I'll learn how to rebase

**Comment by mpolonenko** [comment]
@larsoner all tests have now passed except codecov...not sure why _tracker.py is reducing coverage? Is there something else I should change? Thanks for the help.

**Comment by mpolonenko** [comment]
@larsoner thanks for the link. I’ve made more tests for the tracker and this updated version passed all tests, including coverage.

**Comment by maddycapp27** [comment]
LGTM. Nice addition :)

File: expyfun/stimuli/tests/test_tracker.py [hunk 3]
```diff
@@ -209,7 +209,9 @@ def test_tracker_dealer():
 
     # test TrackerDealer with TrackerBinom
     trackers = [TrackerBinom(None, 0.05, 0.5, 50, stop_early=False)
-                for _ in range(2)]
+                for _ in range(2)]    # start_value scalar type checking
+    pytest.raises(TypeError, TrackerUD, None, 3, 1, [1, 0.5], [1, 0.5], 10,
+                  np.inf, [9, 5], change_indices=[2])
```
[the remark this answers is not in the document]
Comment (mpolonenko) [reply expyfun/stimuli/tests/test_tracker.py]: done! I updated all instances of pystest.raises in the file - for all trackers

File: expyfun/stimuli/tests/test_tracker.py [hunk 4]
```diff
@@ -221,3 +223,62 @@ def test_tracker_dealer():
...
+    rand = np.random.RandomState(0)
+    while not tr.stopped:
+        tr.respond(int(rand.rand() * 100) < tr.x_current)
+
+    tr = TrackerMHW(None, 0, 120)
+    rand = np.random.RandomState(0)
+    while not tr.stopped:
+        tr.respond(int(rand.rand() * 100) < tr.x_current)
+    # test responding after stopped
+    pytest.raises(RuntimeError, tr.respond, 0)
```
[the remark this answers is not in the document]
Comment (mpolonenko) [reply expyfun/stimuli/tests/test_tracker.py]: done

File: expyfun/stimuli/tests/test_tracker.py (hunk 3 above)
[the remark this answers is not in the document]
Comment (larsoner) [reply expyfun/stimuli/tests/test_tracker.py]: My old physics prof used to call this sort of thing "the intrinsic motivator kicking in" :) Thanks for getting all of them

File: expyfun/stimuli/tests/test_tracker.py [hunk 5]
```diff
@@ -205,19 +227,107 @@ def test_tracker_dealer():
     # bad rand type
     trackers = [TrackerUD(None, 1, 1, 0.06, 0.02, 20, 50, 1)
                 for _ in range(2)]
-    pytest.raises(TypeError, TrackerDealer, trackers, rand=1)
+    with pytest.raises(TypeError, match="missing 1 required positional"):
```
Comment (larsoner) [inline expyfun/stimuli/tests/test_tracker.py]: Needs to be more general, maybe just `match='argument'`?

https://travis-ci.org/LABSN/expyfun/jobs/599778799#L1538

Comment (mpolonenko) [reply expyfun/stimuli/tests/test_tracker.py]: Right - I wasn't sure how much to put for this one and it seemed to pass locally when I ran the pytest. I've updated it to just match "argument"

Comment (mpolonenko) [reply expyfun/stimuli/tests/test_tracker.py]: Thanks for catching all theses - sorry there have been so many revisions. First time doing this!

Comment (larsoner) [reply expyfun/stimuli/tests/test_tracker.py]: No problem, all part of the process -- thanks for sticking with it

Comment (mpolonenko) [reply expyfun/stimuli/tests/test_tracker.py]: It's been a fun project!

**Comment by larsoner** [comment]
Thanks @mpolonenko !

**Pull request merged**